xbmc/games/addons/GameClientMouse.cpp

```diff
@@ -114,7 +112,7 @@ void CGameClientMouse::OnButtonRelease(const std::string& button)
 
   event.type                   = GAME_INPUT_EVENT_DIGITAL_BUTTON;
   event.port                   = GAME_INPUT_PORT_MOUSE;
-  event.controller_id          = CONTROLLER_NAME;
+  event.controller_id          = MOUSE_CONTROLLER_ID;
   event.feature_name           = button.c_str();
```

Comment:
both are undesirable. we need to pass the controller's context to this class somehow. See `CGameClientInput::ControllerID()` and notice how the context comes from CGameClient. `CGameClientInput::ControllerID()` is then used to transfer the context to the port manager, which opens a port using that controller ID.